core/src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala

```diff
@@ -841,4 +847,43 @@ object ConvertToDeltaCommand {
     // Allow partition column name starting with underscore and dot
     DeltaFileOperations.defaultHiddenFileFilter(fileName) && !fileName.contains("=")
   }
+
+  def getParquetTable(
+      spark: SparkSession,
+      targetDir: String,
+      catalogTable: Option[CatalogTable],
+      partitionSchema: Option[StructType]): ConvertTargetTable = {
+    val qualifiedDir = ConvertToDeltaCommand.getQualifiedPath(spark, new Path(targetDir)).toString
+    new ParquetTable(spark, qualifiedDir, catalogTable, partitionSchema)
+  }
+
+  def getIcebergTable(
+      spark: SparkSession,
+      targetDir: String,
+      sparkTable: Option[Table],
+      tableSchema: Option[StructType]): ConvertTargetTable = {
+    try {
+      val clazz = Utils.classForName("org.apache.spark.sql.delta.IcebergTable")
+      if (sparkTable.isDefined) {
+        val constFromTable = clazz.getConstructor(
```

Comment:
nit: Catch the `method not found exception` and throw a proper error message if the required jars need to be present? This doesn't need to be done here, may be in a follow up PR.

Reply:
can do in a follow up.